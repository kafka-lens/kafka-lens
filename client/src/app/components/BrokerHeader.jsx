import React from 'react';

const BrokerHeader = ({ brokerID, brokerURI }) => (
  <>
    <p className="broker-header">
Broker ID:
      {brokerID}
    </p>
    <p className="broker-header">
URI:
      {brokerURI}
    </p>
  </>
);

export default BrokerHeader;
