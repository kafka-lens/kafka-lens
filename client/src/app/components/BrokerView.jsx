import React from 'react';
import BrokerHeader from './BrokerHeader';
import BrokerTopicsView from './BrokerTopicsView';
import '../css/BrokerView.scss';

const BrokerView = ({
  isAlive, openSideBar, brokerId, brokerURI, topics,
}) => {
  const brokerColors = {
    Red: '#DC143C',
    Green: '#90EE90',
  };
  let backgroundColor;
  if (isAlive === false) {
    backgroundColor = brokerColors.Red;
  } else {
    backgroundColor = brokerColors.Green;
  }


  return (
    <div
      className="brokerView"
      role="button"
      tabIndex="0"
      onKeyPress={() => openSideBar(brokerId, topics)}
      onClick={() => openSideBar(brokerId, topics)}
      key={brokerId}
      style={{ backgroundColor }}
    >
      <BrokerHeader brokerID={brokerId} brokerURI={brokerURI} />
      <BrokerTopicsView topics={topics} />
    </div>
  );
};

export default BrokerView;
