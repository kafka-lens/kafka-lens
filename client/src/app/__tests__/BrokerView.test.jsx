import React from 'react';
import { shallow } from 'enzyme';

import BrokerView from '../components/BrokerView';
import BrokerHeader from '../components/BrokerHeader';

describe('BrokerView.js unit tests', () => {
  let wrapper;
  const props = {
    isAlive: false,
    brokerColors: {
      Red: '#DC143C',
      Green: '#90EE90',
    },
  };

  beforeAll(() => {
    wrapper = shallow(<BrokerView {...props} />);
  });

  it('Should render without errors', () => {
    expect(wrapper).toMatchSnapshot();
  });

  // test if broker header renders
  it('Should render <BrokerHeader/>', () => {
    expect(wrapper.exists('BrokerHeader')).toEqual(true);
  });

  // test if broker topic view renders
  it('Should render <BrokerTopicsView/>', () => {
    expect(wrapper.exists('BrokerTopicsView')).toEqual(true);
  });

  it('If Broker is inactive, then color should be red', () => {
    expect(wrapper.find('.brokerView').get(0).props.style).toHaveProperty(
      'backgroundColor',
      props.brokerColors.Red,
    );
  });

  it('If Broker is active, then color should be green', () => {
    wrapper.setProps({
      isAlive: true,
    });
    expect(wrapper.find('.brokerView')).toHaveLength(1);
    expect(wrapper.find('.brokerView').get(0).props.style).toHaveProperty(
      'backgroundColor',
      props.brokerColors.Green,
    );
  });
});
