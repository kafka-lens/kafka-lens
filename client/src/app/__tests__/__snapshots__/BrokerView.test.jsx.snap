// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BrokerView.js unit tests Should render without errors 1`] = `
<div
  className="brokerView"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  style={
    Object {
      "backgroundColor": "#DC143C",
    }
  }
  tabIndex="0"
>
  <BrokerHeader />
  <BrokerTopicsView />
</div>
`;
